lib/hdb/ext.c

```diff
@@ -571,6 +543,15 @@ hdb_validate_key_rotation(krb5_context context,
     if (!prev_kr)
         return 0;
 
+    if (prev_kr->base_key_kvno == kr->base_key_kvno) {
+        /*
+         * The new base keys can be the same as the old, but must have
+         * different kvnos.  (Well, not must must.  It's a convention for now.)
+         */
+        krb5_set_error_message(context, EINVAL,
+                               "Base key version numbers for KRs must differ");
+        return EINVAL;
+    }
```

Comment:
Not sure this is still pertinent, looks dodgy if we are strict only monotone strictly increasing base_kvnos...